clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -272,7 +270,65 @@ public void testSyncAutocommitRetriedAfterRetriableException(Errors error) {
 
         // We expect that request should have been retried on this sync commit.
         assertExceptionHandling(commitRequestManger, error, true);
-        assertCoordinatorDisconnect(error);
+    }
+
+    @Test
+    public void testCommitSyncThrowsCommitFailedExceptionOnFencedInstanceId() {
+        testCommitSyncFailsWithCommitFailedExceptionOnError(Errors.FENCED_INSTANCE_ID);
+    }
+
+    @Test
+    public void testCommitSyncThrowsCommitFailedExceptionOnUnknownMemberId() {
+        testCommitSyncFailsWithCommitFailedExceptionOnError(Errors.UNKNOWN_MEMBER_ID);
+    }
+
+    private void testCommitSyncFailsWithCommitFailedExceptionOnError(Errors commitError) {
+        CommitRequestManager commitRequestManger = create(false, 100);
+        when(coordinatorRequestManager.coordinator()).thenReturn(Optional.of(mockedNode));
+
+        Map<TopicPartition, OffsetAndMetadata> offsets = Collections.singletonMap(
+            new TopicPartition("topic", 1),
+            new OffsetAndMetadata(0));
+
+        // Send sync offset commit request that fails with an error that is expected to propagate
+        // a CommitFailedException
+        Long expirationTimeMs = time.milliseconds() + retryBackoffMs;
+        CompletableFuture<Void> commitResult = commitRequestManger.addOffsetCommitRequest(offsets, Optional.of(expirationTimeMs), false);
+        completeOffsetCommitRequestWithError(commitRequestManger, commitError);
+        assertFutureThrows(commitResult, CommitFailedException.class);
+    }
+
+    @Test
+    public void testCommitSyncThrowsOffsetMetadataTooLargeException() {
+        // Error with metadata provided by the user should propagate the exception, so they can handle it.
+        testCommitSyncFailsWithErrorException(Errors.OFFSET_METADATA_TOO_LARGE);
+    }
+
+    @Test
+    public void testCommitSyncThrowsInvalidCommitOffsetSizeException() {
+        // Error with data provided by the user should propagate the exception, so they can handle it.
+        testCommitSyncFailsWithErrorException(Errors.INVALID_COMMIT_OFFSET_SIZE);
+    }
+
+    @Test
+    public void testCommitSyncThrowsGroupAuthorizationException() {
+        testCommitSyncFailsWithErrorException(Errors.GROUP_AUTHORIZATION_FAILED);
+    }
+
+    private void testCommitSyncFailsWithErrorException(Errors commitError) {
```

Comment:
This method is very similar to testCommitSyncFailsWithCommitFailedExceptionOnError. Could we share part of the implementation for both cases? For instance, one idea would be to pass the Errors and the exception Exception as params.

Reply:
Agree, missed that. All done and it did simplified a lot.